default/AI/ProductionAI.py

```diff
@@ -20,17 +23,49 @@
                EnumsAI.AIPriorityType.PRIORITY_PRODUCTION_OUTPOST: EnumsAI.AIShipDesignTypes.outpostShip,
                EnumsAI.AIPriorityType.PRIORITY_PRODUCTION_ORBITAL_OUTPOST: EnumsAI.AIShipDesignTypes.outpostBase,
                EnumsAI.AIPriorityType.PRIORITY_PRODUCTION_COLONISATION: EnumsAI.AIShipDesignTypes.colonyShip,
+               EnumsAI.AIPriorityType.PRIORITY_PRODUCTION_ORBITAL_COLONISATION: EnumsAI.AIShipDesignTypes.colonyBase,
                EnumsAI.AIPriorityType.PRIORITY_PRODUCTION_INVASION: EnumsAI.AIShipDesignTypes.troopShip,
                EnumsAI.AIPriorityType.PRIORITY_PRODUCTION_MILITARY: EnumsAI.AIShipDesignTypes.attackShip,
                EnumsAI.AIPriorityType.PRIORITY_PRODUCTION_ORBITAL_DEFENSE: EnumsAI.AIShipDesignTypes.defenseBase,
                EnumsAI.AIPriorityType.PRIORITY_PRODUCTION_ORBITAL_INVASION: EnumsAI.AIShipDesignTypes.troopBase,
-                                        }
+               }
 
 # TODO: dynamic lookup of hull stats
 hullStats = {}
 
 doDoubleShields = False
 
+design_cache = {}  # dict of tuples (rating,pid,designID,cost) sorted by rating and indexed by priority type
+
+
+def find_best_designs_this_turn():
+    """calculates the best designs for each ship class available at this turn."""
```

Comment:
The docstring is a phrase ending in a period.  https://www.python.org/dev/peps/pep-0257/#one-line-docstrings